app.js

```diff
@@ -1,132 +1,132 @@
 app.post('/notes', (req, res) => {
-  const new_title = req.body.title_input /* JSON Object */
-  const new_createdAt = new Date().toJSON().slice(0,10).replace(/-/g,'/');
+  const newTitle = req.body.titleInput /* JSON Object */
+  const newCreatedAt = new Date()
   const newTask = new Task({
-    title: new_title,
-    createdAt: new_createdAt,
+    title: newTitle,
+    createdAt: newCreatedAt,
     finished: false
   })
 
   newTask.save()
     .then((docs) => {
       console.log("Store successfully.")
     }, (err) => {
-      console.error(err)
+      return res.send("DB error: ", err)
     })
-  res.redirect("/notes")
+  return res.redirect("/notes")
 });
 
 /* Edit	: Click Edit to Select specific TASK and direct to the edit page */
 /* @return: {"id":1,"title":"Feedfire","createdAt":"11/13/2018","finished":true} */
 app.get('/notes/:id/edit', (req, res) => {
-  const edit_id = req.params.id
-  Task.findOne({id: edit_id}, (err, task) => {
+  const editId = req.params.id
+  Task.findById(editId, (err, task) => {
     if(err) {
-      console.error(err);
+      return res.send("DB error: ", err)
     }
-    if(task) {
-      res.render('edit', { task: task })
-    }
-    else {
-      res.send("Task not found!")
+    if(!task) {
+      return res.send("Task not found!")
     }
+    return res.render('edit', { task: task })
   })
 });
 
 /* Update: update the information for the selected TASK */
 app.put('/notes/:id', (req, res) => {
   /* Update the data in the db -> using the APIs form mongosses */
-  const edit_id = req.params.id
-  const new_title = req.body.title_input
+  const editId = req.params.id
+  const newTitle = req.body.titleInput
+  const newCreatedAt = new Date()
 
-  Task.findOne({id: edit_id}, (err, task) => {
+  Task.findById(editId, (err, task) => {
     if(err) {
-      console.error(err);
-    }
-    if(task) {
-      /* Update the Task */
-      task.title = new_title
-      task.save((err) => {
-        if(err) console.log(err)
-      })
-      console.log(`Task id: ${edit_id} has been updated.`)
-      res.redirect("/notes")
+      return res.send("DB error: ", err)
     }
-    else {
-      res.send("Task not found!")
+    if(!task) {
+      return res.send("Task not found!")
     }
+    /* Update the Task */
+    task.title = newTitle
+    task.createdAt = newCreatedAt
+    task.save((err) => {
+      if(err) console.log(err)
+    })
+    console.log(`Task id: ${editId} has been updated.`)
+    return res.redirect("/notes")
   })
 });
 
 /* Delete: delete the selected TASK */
 app.delete('/notes/:id', (req, res) => {
-  const selected_id = req.params.id
+  const selectedId = req.params.id
   /* delete the selcted id from db */
-  Task.deleteOne({id: selected_id}, (err, task) => {
+  Task.findByIdAndDelete(selectedId, (err, task) => {
     if(err) {
-      console.error(err);
+      return res.send("DB error: ", err)
     }
   })
-  res.redirect("/notes")
+  return res.redirect("/notes")
```

Comment:
it has the async race conditions with callback function

Reply:
Gocha, I'm going to wrap my return function inside the `findByIdAndDelete()` method to avoid async race conditions. Thanks for pointing it out!